src/main/java/address/ui/Ui.java

```diff
@@ -0,0 +1,34 @@
+package address.ui;
+
+import address.MainApp;
+import address.browser.BrowserManager;
+import address.controller.MainController;
+import address.model.ModelManager;
+import address.util.Config;
+import javafx.scene.control.Alert;
+import javafx.stage.Stage;
+
+/**
+ * The UI of the app.
+ */
+public class Ui {
+    MainController mainController;
+
+    public Ui(MainApp mainApp, ModelManager modelManager, Config config){
+        BrowserManager.initializeJxBrowserEnvironment();
```

Comment:
If i'm not wrong, some resources can't be initialized in the mainApp init(). For example the Java WebView need to be initialize after the stage is created, which is the start() method. I believe @HansAdrian has tried many different location for this init method but can't seems to find a suitable place.

________________________________
From: Damith C. Rajapakse <notifications@github.com>
Sent: Monday, June 27, 2016 11:31:09 AM
To: HubTurbo/addressbook
Cc: Lim You Liang; Mention
Subject: Re: [HubTurbo/addressbook] Added a UI component to front all ui-related code (#299)


In src/main/java/address/ui/Ui.java<https://github.com/HubTurbo/addressbook/pull/299#discussion_r68519041>:

> +import address.MainApp;
> +import address.browser.BrowserManager;
> +import address.controller.MainController;
> +import address.model.ModelManager;
> +import address.util.Config;
> +import javafx.scene.control.Alert;
> +import javafx.stage.Stage;
> +
> +/**
> + * The UI of the app.
> + */
> +public class Ui {
> +    MainController mainController;
> +
> +    public Ui(MainApp mainApp, ModelManager modelManager, Config config){
> +        BrowserManager.initializeJxBrowserEnvironment();


The thread is decided by the calling method. You should be able to push it down to any class as long as the method chain starts with the init method of MainApp which is run on a non-UI thread.

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/HubTurbo/addressbook/pull/299/files/4e7dbc3324816c66d4e5655c086244cbe2d1168e#r68519041>, or mute the thread<https://github.com/notifications/unsubscribe/AIpoED6ljDA1jotyh15E3k44kn_JUdZyks5qP0P9gaJpZM4I-Qes>.